tests/meltano/core/tracking/test_tracker.py

```diff
@@ -1,22 +1,58 @@
+from __future__ import annotations
+
 import json
+import uuid
+from contextlib import contextmanager
+from typing import Any
 from unittest import mock
 
+import pytest
+
+from meltano.core.project import Project
+from meltano.core.project_settings_service import ProjectSettingsService
 from meltano.core.tracking.tracker import TelemetrySettings, Tracker
+from meltano.core.utils import hash_sha256
+
+
+def load_analytics_json(project: Project) -> dict[str, Any]:
+    with open(project.meltano_dir() / "analytics.json") as analytics_json_file:
+        return json.load(analytics_json_file)
+
+
+def check_analytics_json(project: Project) -> None:
+    analytics_json = load_analytics_json(project)
+
+    for key in ("client_id", "project_id"):
+        value = analytics_json[key]
+        assert isinstance(value, str)
+        uuid.UUID(value)
+
+    assert isinstance(analytics_json["send_anonymous_usage_stats"], bool)
+
+
+@contextmanager
+def delete_analytics_json(project: Project) -> None:
+    (project.meltano_dir() / "analytics.json").unlink()
+    try:
+        yield
+    finally:
+        # Recreate `analytics.json`, to avoid causing issues for other tests within the same class
+        # that expect it. Note that `project` is class-scoped.
+        Tracker(project)
```

Comment:
It might be worth having an explicit test of `Tracker.can_track` or `Tracker.send_anonymous_usage_stats` to make sure that `MELTANO_SEND_ANONYMOUS_USAGE_STATS=FALSE`  and friends actually work. 

I realize thats getting tested indirectly by some state change tests but - since that's sort of the lowest level where a "send or don't send" decision happens its probably worth an explicit check that the config opts and envs are working as expected.